src/dotnet/MsbuildProject.cs

```diff
@@ -166,12 +174,97 @@ public static void EnsureAllReferencesExist(List<string> references)
             }
         }
 
+        public IEnumerable<NuGetFramework> GetTargetFrameworks()
```

Comment:
I think this could likely go into an extension method of Project, instead of into this class. This class is already pretty bloated as it is.